operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -708,6 +708,15 @@ func (r *RedpandaReconciler) clusterConfigFor(ctx context.Context, rp *redpandav
 		return nil, errors.WithStack(err)
 	}
 
+	if conf.Warn() != nil {
+		warn := conf.Warn().Error()
+
+		logger := log.FromContext(ctx).WithName(fmt.Sprintf("ClusterReconciler[%T].clusterConfigFor", *rp))
+		logger.V(log.InfoLevel).Info("warning during configuration processing", warn)
+
+		r.EventRecorder.Eventf(rp, "Warning", redpandav1alpha2.EventSeverityError, warn)
```

Comment:
Updating the logger name isn't particularly useful IMO. It's just more text to scroll right of to get the useful part of the log. we'll be searching the log message to find this line if need be.

Additional args should come in`key`, `value` pairs.

I'd also vote to drop the event emission for now. I do think this is a valuable thing to emit as an event but as is this event will be unintelligible and I'm not familiar enough with the event recording interface to recommend a better direction.

```suggestion
		logger := log.FromContext(ctx)
		logger.V(log.InfoLevel).Info("cluster config reification produced warnings", "warnings", warnings)
```